app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.kt

```diff
@@ -0,0 +1,76 @@
+package ar.com.wolox.android.training.ui.login
+
+import android.content.Context
+import android.content.Intent
+import android.text.method.LinkMovementMethod
+import ar.com.wolox.android.R
+import ar.com.wolox.android.training.ui.home.HomeActivity
+import ar.com.wolox.android.training.ui.signup.SignupActivity
+import ar.com.wolox.android.training.utils.onClickListener
+import ar.com.wolox.android.training.utils.onTextChanged
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import kotlinx.android.synthetic.main.fragment_login.*
+
+class LoginFragment : WolmoFragment<BasePresenter<Any>>() {
+
+    override fun layout(): Int = R.layout.fragment_login
+
+    override fun init() {
+        val sharedPref = activity?.getPreferences(Context.MODE_PRIVATE) ?: return
+
+        val vUserName = sharedPref.getString("Username", "")
+        if (vUserName != null && vUserName.isNotEmpty()) {
```

Comment:
We can change this in the future creating a new Activity called RootActivity. The only purpose of that is going to do this check and redirect the user to the HomeActivity or the Login Activity

Reply:
solved in login-backend-validation (https://github.com/wolox-training/ph-android/pull/5)